src/Message/PurchaseRequest.php

```diff
@@ -70,11 +70,14 @@ private function sign($params, $api_key) {
 
 	public function sendData($data)
     {
+		// FIXME: You're not actually sending any data here.
```

Comment:
OmniPay works on a paradigm of sending and receiving messages. The application sets up messages, "sends" them, then gets a message in return. This runs right through the system, regardless of whether a message is actually sent to the remote gateway or not.

The way I think about it, is that the message is "sent" to the core of OmniPay, and the application does not concern itself with the mechanics of where that message *really* goes, if anywhere.

It's not ideal, because it is some basic understanding that isn't explained well. I think maybe the "send" invoked on the messages by the application should be thought of as "action" - so here is a message in the flow of this payment, go *action* it and let me know the result.